### PR TITLE
change maplibre library and improve map layout #29

### PR DESCRIPTION
**Pull request by koilebeit**
# Pull request

## Proposed changes

change maplibre library and improve map layout #29
## Types of changes

- [ ] New feature (non-breaking change which adds functionality).
- [x] Enhancement (non-breaking change which enhances functionality)
- [ ] Bug Fix (non-breaking change which fixes an issue).
- [ ] Breaking change (fix or feature that would cause existing functionality to change).

## Checklist

- [x] I have read the **[README](./README.md)** document.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

## Summary by CodeRabbit

- **New Features**
	- Added interactive maps with custom markers and cluster visualization for collaborators.
- **Enhancements**
	- Improved map controls and interactions, including click and hover effects on collaborators and clusters.
- **Refactor**
	- Replaced the `MapLibre` component with a more flexible map container setup.
- **Removed**
	- Removed the `MapLibre` component and its configurations to streamline map integration.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -65,5 +65,8 @@
 		"vitest": "^1.2.2",
 		"xml2js": "^0.6.2"
 	},
-	"type": "module"
+	"type": "module",
+	"dependencies": {
+		"maplibre-gl": "^4.0.0"
```
Comment (maehr) [inline package.json:70]: @koilebeit clould be moved to dev-dependencies

File: src/routes/partner/+page.svelte [hunk 2]
```diff
@@ -1,15 +1,7 @@
...
-	import {
-		CircleLayer,
-		GeoJSON,
-		MapLibre,
-		MarkerLayer,
-		Popup,
-		Control,
-		NavigationControl,
-		ScaleControl
-	} from 'svelte-maplibre';
```
Comment (maehr) [inline src/routes/partner/+page.svelte:11]: @koilebeit remove svelte-maplibre via `pnpm remove svelte-maplibre`

File: src/routes/partner/+page.svelte [hunk 3]
```diff
@@ -24,22 +16,58 @@
...
-		});
-	}
+	let maxLength = 0; // default maxLength
+
+	class SelectInputControl {
+		constructor(features, map) {
+			this._map = map;
+			this._features = features;
+
+			// Erstelle das Control-Element
```
Comment (maehr) [inline src/routes/partner/+page.svelte:26]: @koilebeit Please use English for inline comments

File: src/routes/partner/+page.svelte [hunk 4]
```diff
@@ -24,22 +16,58 @@
...
+				option.value = index;
+				option.textContent = feature.properties.name;
+				select.appendChild(option);
+			});
 
-	function handleSelectChange(event) {
-		const selectedFeature = features.find((f) => f.label === event.target.value);
-		if (selectedFeature) {
-			flyToFeature(selectedFeature);
+			// Füge einen Event-Handler zum Zoomen hinzu, wenn ein Feature ausgewählt wird
```
Comment (maehr) [inline src/routes/partner/+page.svelte:42]: @koilebeit Please use English for inline comments

File: src/routes/partner/+page.svelte [hunk 5]
```diff
@@ -51,84 +79,179 @@
...
-		if (str.length > maxLength) {
-			return str.slice(0, maxLength) + '...';
-		} else {
-			return str;
-		}
-	}
+		map = new maplibregl.Map({
+			container: 'map',
+			style:
+				'https://vectortiles.geo.admin.ch/styles/ch.swisstopo.leichte-basiskarte.vt/style.json', //'https://basemaps.cartocdn.com/gl/positron-gl-style/style.json',
```
Comment (maehr) [inline src/routes/partner/+page.svelte:86]: @koilebeit Please remove unused values like https://basemaps.cartocdn.com/gl/positron-gl-style/style.json'

File: src/routes/partner/+page.svelte [hunk 6]
```diff
@@ -51,84 +79,179 @@
...
+					type: 'circle',
+					source: 'collaborators',
+					filter: ['has', 'point_count'],
+					paint: {
+						// Use step expressions (https://maplibre.org/maplibre-style-spec/#expressions-step)
+						// with three steps to implement three types of circles:
+						//   * Blue, 20px circles when point count is less than 100
+						//   * Yellow, 30px circles when point count is between 100 and 750
+						//   * Pink, 40px circles when point count is greater than or equal to 750
+						'circle-color': '#70416C',
```
Comment (maehr) [inline src/routes/partner/+page.svelte:136]: @koilebeit please ask @mtwente if we are still doing `#000080`

File: src/routes/partner/+page.svelte [hunk 7]
```diff
@@ -51,84 +79,179 @@
...
+				});
+
+				map.addLayer({
+					id: 'cluster-count',
+					type: 'symbol',
+					source: 'collaborators',
+					filter: ['has', 'point_count'],
+					layout: {
+						'text-field': '{point_count_abbreviated}',
+						'text-font': ['Frutiger Neue Bold'],
```
Comment (maehr) [inline src/routes/partner/+page.svelte:148]: @koilebeit Please do use our default font 'Euclid Circular B' as defined in https://github.com/Stadt-Geschichte-Basel/stadtgeschichtebasel.ch/blob/main/src/app.postcss

File: src/routes/partner/+page.svelte [hunk 8]
```diff
@@ -51,84 +79,179 @@
...
+					id: 'collaborators',
+					type: 'symbol',
+					source: 'collaborators',
+					filter: ['!', ['has', 'point_count']],
+					layout: {
+						'icon-image': 'custom-marker',
+						'icon-overlap': 'always',
+						'icon-size': 1,
+						'text-field': ['get', 'label'],
+						'text-font': ['Frutiger Neue Bold'],
```
Comment (maehr) [inline src/routes/partner/+page.svelte:168]: @koilebeit Please do use our default font 'Euclid Circular B' as defined in https://github.com/Stadt-Geschichte-Basel/stadtgeschichtebasel.ch/blob/main/src/app.postcss

File: src/routes/partner/+page.svelte [hunk 9]
```diff
@@ -51,84 +79,179 @@
...
-				>
-			</MarkerLayer>
-		</GeoJSON>
-	</MapLibre>
+<section class="h-full w-full">
+	<div class="h-full w-full" id="map"></div>
 </section>
+
+<style>
+	:global(body) {
```
Comment (maehr) [inline src/routes/partner/+page.svelte:254]: @koilebeit body has a big scope. let's discuss this in the next review

File: src/lib/data/pin-48.png
Comment (maehr) [inline src/lib/data/pin-48.png:1]: @koilebeit This is not data, please move it either to static or create a new folder (see <https://stackoverflow.com/questions/71789244/where-to-put-images-with-sveltekit>)

File: src/routes/partner/+page.svelte (hunk 7 above)
Comment (koilebeit) [reply src/routes/partner/+page.svelte:148]: Only fonts that are defined as glyphs in the style (https://vectortiles.geo.admin.ch/styles/ch.swisstopo.leichte-basiskarte.vt/style.json) can be used.

From a cartographic point of view, this also makes sense.

Whether and how custom fonts can be integrated is not yet clear to me.

File: src/routes/partner/+page.svelte (hunk 8 above)
Comment (koilebeit) [reply src/routes/partner/+page.svelte:168]: Only fonts that are defined as glyphs in the style (https://vectortiles.geo.admin.ch/styles/ch.swisstopo.leichte-basiskarte.vt/style.json) can be used.

From a cartographic point of view, this also makes sense.

Whether and how custom fonts can be integrated is not yet clear to me.

Comment (maehr) [reply src/routes/partner/+page.svelte:168]: Makes total sense! Sorry for bringing it up

File: src/routes/partner/+page.svelte [hunk 10]
```diff
@@ -51,84 +79,171 @@
...
+				});
+
+				map.addLayer({
+					id: 'cluster-count',
+					type: 'symbol',
+					source: 'collaborators',
+					filter: ['has', 'point_count'],
+					layout: {
+						'text-field': '{point_count_abbreviated}',
+						'text-font': ['Frutiger Neue Bold'],
```
[the remark this answers is not in the document]
Comment (maehr) [reply src/routes/partner/+page.svelte:148]: Only frutiger and some other standard fonts can be used per docs.

File: src/routes/partner/+page.svelte [hunk 11]
```diff
@@ -51,84 +264,39 @@
...
+			container: 'map',
+			style: mapProperties.style,
+			center: mapProperties.center,
+			zoom: mapProperties.zoom,
+			maxBounds: mapProperties.maxBounds,
+			minZoom: mapProperties.minZoom,
+			maxZoom: mapProperties.maxZoom,
+			scrollZoom: mapProperties.scrollZoom,
+			attributionControl: mapProperties.attributionControl
+		});
```
[the remark this answers is not in the document]
Comment (maehr) [reply src/routes/partner/+page.svelte:278]: This seems to be overkill

File: src/routes/partner/+page.svelte (hunk 10 above)
[the remark this answers is not in the document]
Comment (koilebeit) [reply src/routes/partner/+page.svelte:148]: Only fonts that are defined as glyphs in the style (https://vectortiles.geo.admin.ch/styles/ch.swisstopo.leichte-basiskarte.vt/style.json) can be used.

File: src/routes/partner/+page.svelte [hunk 12]
```diff
@@ -51,84 +79,171 @@
...
+					id: 'collaborators',
+					type: 'symbol',
+					source: 'collaborators',
+					filter: ['!', ['has', 'point_count']],
+					layout: {
+						'icon-image': 'custom-marker',
+						'icon-overlap': 'always',
+						'icon-size': 1,
+						'text-field': ['get', 'label'],
+						'text-font': ['Frutiger Neue Bold'],
```
[the remark this answers is not in the document]
Comment (koilebeit) [reply src/routes/partner/+page.svelte:168]: Only fonts that are defined as glyphs in the style (https://vectortiles.geo.admin.ch/styles/ch.swisstopo.leichte-basiskarte.vt/style.json) can be used.

File: src/routes/partner/+page.svelte (hunk 6 above)
Comment (koilebeit) [reply src/routes/partner/+page.svelte:153]: maybe we'll wait until the colors are really finalized?

**Comment by maehr** [comment]
@koilebeit I suggest you copy `package.json` from main, run `pnpm remove svelte-maplibre && pnpm add maplibre-gl -D` and commit `package.json` and `pnpm-lock.yml` to this PR.

**Comment by koilebeit** [comment]
> @koilebeit I suggest you copy `package.json` from main, run `pnpm remove svelte-maplibre && pnpm add maplibre-gl -D` and commit `package.json` and `pnpm-lock.yml` to this PR.

Thanks for the help. Maybe one day it will work...? Now there is a lint-
Error:   
```
32:6  error  'maxLength' is assigned a value but never used  no-unused-vars
```
But maxLength is used and without the var the map will not show up

**Comment by koilebeit** [comment]
> Thanks again. Please use https://fontawesome.com/icons/location-dot?f=classic&s=solid instead of the `pin-48.png` if possible. The credits are embedded and vector graphics are scalable. Afterwards we are ready to go.

As far as I can see, there is no native way to add your own SVG to the map, except to convert it to an image first. There are maplibre plugins that can do this, but to avoid version complications between maplibre and the plugin, I suggest sticking with the PNG for now so that the map works stable.
see also 

- [https://github.com/maplibre/maplibre/discussions/175](https://github.com/maplibre/maplibre/discussions/175)
- [https://github.com/nationalparkservice/npmap5-plugins/tree/main/svg-icons](https://github.com/nationalparkservice/npmap5-plugins/tree/main/svg-icons) or 
- [https://github.com/rbrundritt/maplibre-gl-svg](https://github.com/rbrundritt/maplibre-gl-svg)

**Review by maehr — APPROVED** [review]

**Pull request merged**